pyMOE/aperture.py

```diff
@@ -9,6 +9,38 @@
 import numpy as np
 
 class Aperture:
+    """
+    Class Aperture:
+        Creates an Aperture object that is an homogenous array of values corresponding
+        to the transfer function matrix across the aperture
+    
+    Args:
+        x = Vector for the x axis
+        y = Vector for the y axis
+    
+    Methods:
+        aperture: returns the aperture
+        shape: returns the shape of the aperture
+
+    """
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+        self.XX, self.YY = np.meshgrid(x, y)
+        self.pixel_x = self.x[1]-self.x[0]
+        self.pixel_y = self.y[1]-self.y[0]
+        
+
+        self.aperture = np.zeros(self.XX.shape)
+
+    @property
+    def shape(self):
+        return self.aperture.shape
+
+
+
+
+class ApertureField:
```

Comment:
The ApertureField is not really used at this time.
The idea would be the Aperture is a real array which can be an amplitude or a phase, and the user would decide. For example, a circular opening would have 0 ampltude everywhere except in the circular region where it would be 1. A continuous opening transmission array could also be represented by a continuous variation of the amplitude. On the other hand, the array could correspond to a phase if the user so selects, with a variation between ±π or any other phase progression. It is up to the user.

For the ApertureField or the Field class, I would consider using a similar class to aperture, but instead of a real valued array, it is a complex array corresponding to the complex transmission function. The Field class would have internal property getters to extract the amplitude (as abs(field)) and the phase (as angle(field)). 
However, handling the Field class would require more care in the generator functions, so it was put on hold at this time, and left partially written inside aperture.py.

To illustrate with an example: consider we are doing a 1. Fresnel Zone Plate [moe.generate.fresnel_zone_plate_aperture] (which is an aperture/opening plate with rings of amplitude 0 and 1) or a 2. Fresnel lens phase [moe.generate.fresnel_phase] (which is a truncated/discretized function of the phase shift along the optical path of the lens). 
Using the Aperture class, the generator function does not care if it is to input the calculated array into the amplitude or the phase, it goes straight into the aperture and the user selects the appropriate usage of this array.
Using the Field class, the functions would need to know, or have the argument passed, where the calculated array must be stored. 
For mask generation, using the Aperture class is more straightforward and less prone to user misunderstandings if it is an amplitude or a phase.

However, one advantage of using the Field class, and a requirement in the future, is for the propagation kernels. I would suggest that for the propagation we take as input the Field class with a complex field array instead of the Aperture class. For this reason, in the future we will need to make a decision of either keeping Aperture and Field classes separate, or joining them and using a single class for everything. The latter option requires a slightly higher learning curve to use the library.

Reply:
Suggest to add an issue regarding this, with enhancement tag for future reference.